enhancements/authentication/service-ca-cert-generation-for-statefulset-pods.md

```diff
@@ -0,0 +1,444 @@
+---
+title: service-ca-cert-generation-for-statefulset-pods
+authors:
+  - "@mtrmac"
+reviewers:
+  - TBD
+approvers:
+  - TBD
+creation-date: 2021-03-01
+last-updated: 2021-03-01
+status: implementable
+see-also:
+replaces:
+superseded-by:
+---
+
+# Service CA Certificate Generation for StatefulSet Pods
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+The Service CA can be used to automatically generate certificates for StatefulSets,
+to allow TLS-protected communication between StatefulSet pods.
+
+The user adds a `service.beta.openshift.io/serving-cert-secret-name: $secretName` annotation,
+and a `$secretName` secret is generated, which can be mounted in the StatefulSet’s pods.
```

Comment:
Currently the private key is PKCS#1 which only supports RSA and is no longer supported by some applications (eg. logstash). Can this be upgraded to PKCS#8 please?

Reply:
@jackevans43 That seems well out of scope of _this_ change. Please file a separate issue at https://github.com/openshift/service-ca-operator/ , or a separate enhancement proposal.